ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractMiningCoordinator.java

```diff
@@ -63,75 +72,116 @@ public AbstractMiningCoordinator(
       final BlockHeader parentHeader,
       final List<Transaction> transactions,
       final List<BlockHeader> ommers) {
-    M miner = executor.createMiner(parentHeader);
+    final M miner = executor.createMiner(Subscribers.none(), parentHeader);
     return Optional.of(miner.createBlock(parentHeader, transactions, ommers));
   }
 
   @Override
-  public void enable() {
+  public void start() {
     synchronized (this) {
-      if (isEnabled) {
+      if (state != State.IDLE) {
+        return;
+      }
+      state = State.RUNNING;
+      startMiningIfPossible();
+    }
+  }
+
+  @Override
+  public void stop() {
+    synchronized (this) {
+      if (state != State.RUNNING) {
         return;
       }
-      if (syncState.isInSync()) {
-        startAsyncMiningOperation();
+      state = State.STOPPED;
+      haltCurrentMiningOperation();
+      executor.shutDown();
+    }
+  }
+
+  @Override
+  public void awaitStop() throws InterruptedException {
+    executor.awaitShutdown();
+  }
+
+  @Override
+  public boolean enable() {
+    synchronized (this) {
+      if (isEnabled) {
+        return true;
       }
       isEnabled = true;
+      startMiningIfPossible();
     }
+    return true;
   }
 
   @Override
-  public void disable() {
+  public boolean disable() {
     synchronized (this) {
       if (!isEnabled) {
-        return;
+        return false;
       }
-      haltCurrentMiningOperation();
       isEnabled = false;
+      haltCurrentMiningOperation();
     }
+    return false;
   }
 
   @Override
-  public boolean isRunning() {
+  public boolean isMining() {
     synchronized (this) {
       return currentRunningMiner.isPresent();
     }
   }
 
-  protected void startAsyncMiningOperation() {
+  private synchronized boolean startMiningIfPossible() {
+    if ((state != State.RUNNING) || !isEnabled || !syncState.isInSync() || isMining()) {
+      return false;
+    }
+
+    startAsyncMiningOperation();
+    return true;
+  }
+
+  private void startAsyncMiningOperation() {
     final BlockHeader parentHeader = blockchain.getChainHeadHeader();
-    currentRunningMiner = Optional.of(executor.startAsyncMining(minedBlockObservers, parentHeader));
+    currentRunningMiner = executor.startAsyncMining(minedBlockObservers, parentHeader);
   }
 
-  protected void haltCurrentMiningOperation() {
-    currentRunningMiner.ifPresent(M::cancel);
+  private synchronized boolean haltCurrentMiningOperation() {
```

Comment:
nit: seems the return value is only used for logging - maybe the existence of the miner could be used ... or not ... I hate threads.

Reply:
That particular logging statement is valuable to node operators.